File: Beakjoon/Python/1260 DFS와 BFS.py
from collections import deque

n,m,v= map(int , input().split())

visited = [False for i in range(n+1)]
graph = [[] for i in range(n+1)]
for i in range(m) :
    st, ed = map(int, input().split())
    graph[st].append(ed)
    graph[ed].append(st)
for info in graph :
    info.sort()
    
def dfs(start) :
    stack = []
    stack.append(start)
    visited[start] = True
    print(start, end = " ")
    
    while stack :
        v = stack[-1]
        cantmove = True
        for t in graph[v] :
            if visited[t] == False :
                stack.append(t)
                visited[t] = True
                print(t, end = " ")
                cantmove = False
                break
        
        if cantmove :
            stack.pop()

def bfs(start) :
    queue = deque()
    queue.append(start)
    visited[start] = True
    print(start, end = " ")
    
    while queue :
        v = queue.popleft()
        for t in graph[v] :
            if visited[t] == False :
                queue.append(t)
                visited[t] = True
                print(t, end = " ")

dfs(v)
print()
visited = [False for i in range(n+1)]
bfs(v)

File: Beakjoon/Javascript/1260 DFS와 BFS.js
const arr = require("fs")
    .readFileSync("/dev/stdin")
    .toString()
    .trim()
    .split("\n");

const [n, m, v] = arr[0].split(" ").map((el) => +el);
let answer = [];
const graph = {};

for (let i = 1; i <= n; i++) {
    graph[i] = [];
}
for (let i = 1; i <= m; i++) {
    const [f, t] = arr[i].split(" ").map((el) => +el);
    graph[f].push(t);
    graph[t].push(f);
}

Object.keys(graph).forEach((el) => graph[String(el)].sort((a, b) => a - b));

let visited = Array.from({length: n + 1}, () => false);
const init = () => {
    visited = Array.from({length: n + 1}, () => false);
    answer = [];
};

const dfs = (graph, visited, vt) => {
    if (visited[vt]) return;
    visited[vt] = true;
    answer.push(vt);
    graph[String(vt)].forEach((el) => dfs(graph, visited, el));
};
const bfs = (graph, visited, start) => {
    visited[start] = true;
    const queue = [];
    queue.push(start);

    while (queue.length !== 0) {
        const v = queue.shift();
        answer.push(v);
        graph[String(v)].forEach((el) => {
            if (visited[el]) return;
            queue.push(el);
            visited[el] = true;
        });
    }
};

dfs(graph, visited, v);
console.log(answer.join(" "));

init();
bfs(graph, visited, v);
console.log(answer.join(" "));
